src/Graph.cpp

```diff
@@ -1,17 +1,48 @@
 #include "Graph.h"
+/*
+#include <algorithm>
+#include <queue>
+#include <iostream>
+#include <cfloat>
 #include <string>
 #include <fstream>
 #include <vector>
 #include <utility>
 #include <stdexcept>
 #include <sstream>
-#include <iostream>
+*/
+
+std::vector<Node*>* Graph::getNodes() {
+    return &nodes_;
+}
+
+unsigned int Graph::getSize() {
+    return nodes_.size();
+}
+
+Node *Graph::getNode(std::string zone)
+{
+    for (unsigned i = 0; i < nodes_.size(); ++i)
+    {
+        if (nodes_.at(i)->getZone() == zone)
+        {
+            return nodes_.at(i);
+        }
+    }
+    return NULL;
+}
 
 Graph::Graph(std::string filename)
 {
     buildGraph(filename);
 }
 
+Graph::Graph(std::vector<Node*> nodes) {
+    nodes_ = nodes;
+}
+
+Graph::Graph() {}
```

Comment:
please remove. this is not necessary

Reply:
The graph vector function is used for a current Dijkstra test case. Will remove once Dijkstra tests are completed. Added a TODO

Reply:
ahhh gotcha